pkg/apis/v1beta1/ec2nodeclass_status.go

```diff
@@ -68,4 +70,23 @@ type EC2NodeClassStatus struct {
 	// InstanceProfile contains the resolved instance profile for the role
 	// +optional
 	InstanceProfile string `json:"instanceProfile,omitempty"`
+	// Conditions contains signals for health and readiness
+	// +optional
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+
+var (
+	// 	NodeClassReady = "Ready" condition indicates that subnets, security groups, AMIs and instance profile for nodeClass were resolved
+	NodeClassReady = "Ready"
```

Comment:
```suggestion
	ConditionTypeNodeClassReady = "Ready"
```

We should prefix this with condition type so that all condition types are classified like this. We should make the same change over in the upstream repo too